src/asm/opt.c

```diff
@@ -71,16 +78,31 @@ void opt_Parse(char *s)
 			error("Must specify exactly 4 characters for option 'g'\n");
 		break;
 
+	case 'f':
+		if (strlen(&s[1]) <= 2) {
```

Comment:
Rather than `opt fN`, `opt q.N` would be clearer that `N` is `y` in `Qx.y` (also `Q` instead of `f`, which makes more sense for floating-point)

Reply:
Agreed about `Q` vs `f`, but adding a `.` before it looks unusual and awkward. If we allowed specifying *m* and/or *n* in `Qm.n` it would make sense, but here `Q` is just another option taking a single numeric argument.

Reply:
This is another case of "it's impossible to get it wrong".
Option letters are completely obscure without reading the documentation, and the documentation would clearly indicate what the option means. Either the user has read the docs (in which case they know what the option does) or they haven't (in which case they have no idea of what any flag does).
Guessing at what an option does is not a possibility because options, being single letters, carry the least amount of information possible in their names.

Reply:
It doesn't look quite awkward, since it's just the fractional part of `Qx.y`. And since either can be derived from the other, since the width is already fixed, it need not be specified. Additionally, specifying the precision makes the most sense, also if we are to eventually allow greater widths.

However, it is not clear which one it is as a single number, but the dot makes it more obvious (and also parallels C and RGBDS' usage e.g. format specifiers.

The argument that "you can't know it without reading the docs first" makes no sense, as it's not about that, but instead making the syntax more intuitive / self-documenting, and thus easier to remember. You can have read the docs and yet have no idea—that's why we don't have INTERCAL flags, but e.g. `p` for **p**adding.

Reply:
In "`-Q 8`" or `opt Q8`, the 8 is clearly the fractional precision, because as the documentation would explain, that's the only precision we allow to be configured. Numbers are internally 32-bit, so the integral precision is implied.

I do understand that `opt Q.8` looks more like the "Q24.8" notation, but it makes people type a "." that has no actual effect on the result. It's not as if `opt Q8` vs `opt Q.8` would do anything different.

Reply:
Similarly, whitespace is not significant, nor are commas to separate operands in instructions; however, they are more readable. So there is significance in technically-not-required punctuation.

Reply:
So, optionally allow a single `.` before the number? Sure.